CrowsZA/functions/fn_fireSupport.sqf

```diff
@@ -0,0 +1,205 @@
+/*/////////////////////////////////////////////////
+Author: Radkewolf
+			   
+File: fn_fireSupport.sqf
+Parameters: pos
+Return: none
+
+Sets off firesupport in the targeted area
+
+*///////////////////////////////////////////////
+params [["_pos",[0,0,0],[[]],3], ["_logic",objNull,[objNull]]];
+
+// Validate module target
+scopeName "Main";
+private _fnc_errorAndClose = {
+    params ["_msg"];
+    diag_log _msg;
+    breakOut "Main";
+};
+
+// open dialog
+//ZEN
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+	_dialogResult params
+	[
+		"_type",
+		"_customType",
+		"_radius",
+		"_seconds",
+		"_delay",
+		"_salvos",
+		"_guns"
+	];
+
+	//Get in params again
+	_in params [["_pos",[0,0,0],[[]],3], ["_logic",objNull,[objNull]]];
+
+	//Check if placed on an object
+	_create = isNull _logic;
+
+	//Check if object is firesupport module
+	if(!_create)
+	then
+	{
+		if(!(_logic isKindOf "Logic"))
```

Comment:
instead of two nested if, could we combine the ifs? Makes it a bit easier to see the flow. And as I understand it does sequential checks, so it won't error if ``_logic`` is null as the first condition catch that. Although would have to test it to be sure. :) 

```sqf
	// by default we create
	private _create = true;

	if (!isNull _logic && _logic isKindOf "logic" && (_logic getVariable ["crowsZA_module",""]) == "firesupport") then {
		_create = false;
	};
```

Reply:
I tried that, but it didn't work. I can give it another try. Same reason why it reads the variable first before the if-statement a few lines below.

Reply:
oh alright. If you tried I am probably lacking some part of understanding of it. That's fine. If you want you can give it one go again, otherwise just keep it as is.